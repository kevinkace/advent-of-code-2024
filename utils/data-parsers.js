const EOL = "\n";
const DELIM = "   ";

export function splitToLists(data, { eol = EOL, delim = DELIM } = {}) {
    return data.split(eol).map(line => line.split(delim)).reduce((acc, cur) => {
        acc[0].push(Number(cur[0]));
        acc[1].push(Number(cur[1]));

        return acc;
    }, [[], []]);
}
