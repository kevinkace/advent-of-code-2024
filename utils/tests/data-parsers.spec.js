import { expect, test } from "vitest";

import { splitToLists } from "../data-parsers";

import example1 from "../../01/data/example-1";

test("splitToLists", () => {
    expect(splitToLists(example1)).toMatchSnapshot();
})