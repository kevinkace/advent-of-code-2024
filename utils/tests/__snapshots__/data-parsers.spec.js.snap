// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse2d 1`] = `
[
  [
    7,
    6,
    4,
    2,
    1,
  ],
  [
    1,
    2,
    7,
    8,
    9,
  ],
  [
    9,
    7,
    6,
    2,
    1,
  ],
  [
    1,
    3,
    2,
    4,
    5,
  ],
]
`;

exports[`splitToLists 1`] = `
[
  [
    3,
    4,
    2,
    1,
    3,
    3,
  ],
  [
    4,
    3,
    5,
    3,
    9,
    3,
  ],
]
`;
